flutter/lib/src/jvm/jvm_exception.dart

```diff
@@ -0,0 +1,113 @@
+
+const _emptyString = '';
+const _causedBy = 'Caused by:';
+const _suppressed = 'Suppressed:';
+const _newLine = '\n';
+
+class JvmException {
+  JvmException({
+    this.thread,
+    this.type,
+    this.description,
+    required this.stackTrace,
+    this.causes,
+    this.suppressed,
+  });
+
+  factory JvmException.parse(String exception) {
+    return _parse(exception);
+  }
+
+  static JvmException? tryParse(String exception) {
+    try {
+      return JvmException.parse(exception);
+    } catch (_) {
+      return null;
+    }
+  }
+
+  final String? thread;
+  final String? type;
+  final String? description;
+
+  final List<JvmFrame> stackTrace;
+
+  final List<JvmException>? causes;
+  final List<JvmException>? suppressed;
+
+  static JvmException _parse(String string) {
+    final lines = string.split(_newLine);
+    final first = _parseFirstLine(lines.first);
+    lines.removeAt(0);
+
+    final thisException = <String>[];
+    final causes = <List<String>>[];
+    final supressed = <List<String>>[];
+
+    var frames = thisException;
+    for (final line in lines) {
+      var trimmed = line.trim();
+
+      if (trimmed.startsWith(_causedBy)) {
+        trimmed = trimmed.replaceAll(_causedBy, _emptyString).trim();
+        causes.add(<String>[]);
+        frames = causes.last;
+      }
+      if (trimmed.startsWith(_suppressed)) {
+        trimmed = trimmed.replaceAll(_suppressed, _emptyString).trim();
+        supressed.add(<String>[]);
+        frames = supressed.last;
+      }
+      frames.add(trimmed);
+    }
+
+    final thisExceptionFrames =
+        thisException.map((e) => JvmFrame.parse(e)).toList(growable: false);
+
+    final suppressedExceptions = supressed
+        .map((e) => JvmException.parse(e.join(_newLine)))
+        .toList(growable: false);
+
+    final causeExceptions = causes
+        .map((e) => JvmException.parse(e.join(_newLine)))
+        .toList(growable: false);
+
+    return JvmException(
+      thread: first[0],
+      type: first[1],
+      description: first[2],
+      stackTrace: thisExceptionFrames,
+      suppressed: suppressedExceptions.isEmpty ? null : suppressedExceptions,
+      causes: causeExceptions.isEmpty ? null : causeExceptions,
+    );
+  }
+
+  // first element is the thread
+  // second element is the Exception type
+  // thirs element is the exception description
+  static List<String?> _parseFirstLine(String firstLine) {
+    final list = <String?>[];
+    // Exception in thread "main" java.lang.Exception: Main block
+    firstLine = firstLine.trim();
+    if (firstLine.startsWith('Exception in thread "')) {
+      firstLine = firstLine.replaceFirst('Exception in thread "', '');
+      final firstLineSplitted = firstLine.split('"');
+      list.add(firstLineSplitted.first);
+      firstLine = firstLineSplitted[1].trim();
+    } else {
+      list.add(null);
+    }
+    final firstLineSplitted = firstLine.split(': ');
+    final type = firstLineSplitted.first;
```

Comment:
Hm, I don't think so. Can you come up with a case where this happens?

There's already a test case with the following exception:
```
java.lang.IllegalArgumentException: Unsupported value: '[Ljava.lang.StackTraceElement;@ba6feed' of type 'class [Ljava.lang.StackTraceElement;'
	at io.flutter.plugin.common.StandardMessageCodec.writeValue(StandardMessageCodec.java:292)
	at io.flutter.plugin.common.StandardMethodCodec.encodeSuccessEnvelope(StandardMethodCodec.java:59)
	at io.flutter.plugin.common.MethodChannel\$IncomingMethodCallHandler\$1.success(MethodChannel.java:267)
	at io.sentry.samples.flutter.MainActivity.configureFlutterEngine\$lambda-0(MainActivity.kt:40)
	at io.sentry.samples.flutter.MainActivity.lambda\$TiSaAm1LIEmKLVswI4BlR_5sw5Y(Unknown Source:0)
	at io.sentry.samples.flutter.-\$\$Lambda\$MainActivity\$TiSaAm1LIEmKLVswI4BlR_5sw5Y.onMethodCall(Unknown Source:2)
	at io.flutter.plugin.common.MethodChannel\$IncomingMethodCallHandler.onMessage(MethodChannel.java:262)
	at io.flutter.embedding.engine.dart.DartMessenger.invokeHandler(DartMessenger.java:296)
	at io.flutter.embedding.engine.dart.DartMessenger.lambda\$dispatchMessageToQueue\$0\$DartMessenger(DartMessenger.java:320)
	at io.flutter.embedding.engine.dart.-\$\$Lambda\$DartMessenger\$TsixYUB5E6FpKhMtCSQVHKE89gQ.run(Unknown Source:12)
	at android.os.Handler.handleCallback(Handler.java:938)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:210)
	at android.os.Looper.loop(Looper.java:299)
	at android.app.ActivityThread.main(ActivityThread.java:8138)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit\$MethodAndArgsCaller.run(RuntimeInit.java:556)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1037)
```

Assuming the else-branch is called:

`firstLine.split(': ')` results in `['java.lang.IllegalArgumentException', 'Unsupported value', "'[Ljava.lang.StackTraceElement;@ba6feed' of type 'class [Ljava.lang.StackTraceElement;'"]` and `firstLineSplitted.first` is then `java.lang.IllegalArgumentException`

Reply:
Well, because we are dealing with string im assuming anything can be passed as parameter. 
Let's say, suddenly we receive the fallowing message 'Some unknown error occurred', this entire message would be used as `type`. 

But also Is not likely java exception format will change any time soon, so I think we are good.

Reply:
True, I hope though, that the Flutter team makes sure the stack trace paramaeter always contains a stacktrace 😅

Reply:
We also have a try/catch within the conversion, so in the last case, it returns the original Event with the given String as a Stack trace.